Source/sdc/Castro_sdc.cpp

```diff
@@ -5,350 +5,375 @@
+            // -- including one ghost cell
+            U_center.resize(bx1, NUM_STATE);
+            ca_make_cell_center(BL_TO_FORTRAN_BOX(bx1),
+                                BL_TO_FORTRAN_FAB(Sborder[mfi]),
+                                BL_TO_FORTRAN_FAB(U_center),
+                                AMREX_INT_ANYD(domain_lo), AMREX_INT_ANYD(domain_hi));
+
+            // sometimes the Laplacian can make the species go negative near discontinuities
+            ca_normalize_species(AMREX_INT_ANYD(bx1.loVect()), AMREX_INT_ANYD(bx1.hiVect()),
+                                 BL_TO_FORTRAN_ANYD(U_center));
+
+            // convert the C source to cell-centers
+            C_center.resize(bx1, NUM_STATE);
+            ca_make_cell_center(BL_TO_FORTRAN_BOX(bx1),
+                                BL_TO_FORTRAN_FAB(C_source[mfi]),
+                                BL_TO_FORTRAN_FAB(C_center),
+                                AMREX_INT_ANYD(domain_lo), AMREX_INT_ANYD(domain_hi));
+
+            // solve for the updated cell-center U using our cell-centered C -- we
+            // need to do this with one ghost cell
+            U_new_center.resize(bx1, NUM_STATE);
+
+            // initialize U_new with our guess for the new state, stored as
+            // an average in Sburn
+            ca_make_cell_center(BL_TO_FORTRAN_BOX(bx1),
+                                BL_TO_FORTRAN_FAB(Sburn[mfi]),
+                                BL_TO_FORTRAN_FAB(U_new_center),
+                                AMREX_INT_ANYD(domain_lo), AMREX_INT_ANYD(domain_hi));
+
+            ca_sdc_update_centers_o4(BL_TO_FORTRAN_BOX(bx1), &dt_m,
+                                     BL_TO_FORTRAN_3D(U_center),
+                                     BL_TO_FORTRAN_3D(U_new_center),
+                                     BL_TO_FORTRAN_3D(C_center),
+                                     &sdc_iteration);
+
+            // compute R_i and in 1 ghost cell and then convert to <R> in
+            // place (only for the interior)
+            R_new.resize(bx1, NUM_STATE);
+            Elixir elix_R_new = R_new.elixir();
+            amrex::Array4<const amrex::Real> const& R_new_arr=R_new.array();
+
+            ca_instantaneous_react(BL_TO_FORTRAN_BOX(bx1),
+                                   BL_TO_FORTRAN_3D(U_new_center),
+                                   BL_TO_FORTRAN_3D(R_new));
+
+            ca_make_fourth_in_place(BL_TO_FORTRAN_BOX(bx),
+                                    BL_TO_FORTRAN_FAB(R_new),
+                                    AMREX_INT_ANYD(domain_lo), AMREX_INT_ANYD(domain_hi));
+
+            // now do the conservative update using this <R> to get <U>
+            // We'll also need to pass in <C>
+            ca_sdc_conservative_update(bx, dt_m, k_new_m_start_arr, k_new_m_end_arr,
+                                       C_source_arr,R_new_arr);
+
+        }
+#else
+        amrex::Array4<const amrex::Real> const& k_new_m_start_arr=
```

Comment:
you might consider doing `using namespace amrex` and dropping all of the `amrex::`